website/docs/r/log_analytics_solution.html.markdown

```diff
@@ -0,0 +1,75 @@
+---
+layout: "azurerm"
+page_title: "Azure Resource Manager: azurerm_log_analytics_solution"
+sidebar_current: "docs-azurerm-resource-oms-log-analytics-solution"
+description: |-
+  Creates a new Log Analytics (formally Operational Insights) Solution.
+---
+
+# azurerm_log_analytics_solution
+
+Creates a new Log Analytics (formally Operational Insights) Solution.
+
+## Example Usage
+
+```hcl
+resource "azurerm_resource_group" "test" {
+	name     = "k8s-log-analytics-test"
+	location = "westeurope"
+}
+
+resource "random_id" "workspace" {
+  keepers = {
+    # Generate a new id each time we switch to a new resource group
+    group_name = "${azurerm_resource_group.test.name}"
+  }
+
+  byte_length = 8
+}
+  
+resource "azurerm_log_analytics_workspace" "test" {
+	name                = "k8s-workspace-${random_id.workspace.hex}"
+	location            = "${azurerm_resource_group.test.location}"
+	resource_group_name = "${azurerm_resource_group.test.name}"
+	sku                 = "Free"
+}
+  
+resource "azurerm_log_analytics_solution" "test" {
+	solution_name         = "Containers"
+	location              = "${azurerm_resource_group.test.location}"
+	resource_group_name   = "${azurerm_resource_group.test.name}"
+	workspace_resource_id = "${azurerm_log_analytics_workspace.test.id}"
+	workspace_name        = "${azurerm_log_analytics_workspace.test.name}"
+	
+	plan {
+	  publisher      = "Microsoft"
+	  product        = "OMSGallery/Containers"
+	}
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `solution_name` - (Required) Specifies the name of the solution to be deployed. See [here for options](https://docs.microsoft.com/en-us/azure/log-analytics/log-analytics-add-solutions). Note: Resource tested with only `Container` solution. Changing this forces a new resource to be created.
```

Comment:
Can we remove `Note: Resource tested with only `Container` solution` - we can add additional test-cases as needed